baseplate/sidecars/publisher_queue_utils.py

```diff
@@ -0,0 +1,97 @@
+import contextlib
+
+from typing import Dict
+from typing import Generator
+from typing import Optional
+
+import gevent
+
+from gevent.server import StreamServer
+
+from baseplate.lib import config
+from baseplate.lib.message_queue import InMemoryMessageQueue
+from baseplate.lib.message_queue import MessageQueue
+from baseplate.lib.message_queue import PosixMessageQueue
+from baseplate.lib.message_queue import QueueType
+from baseplate.lib.message_queue import RemoteMessageQueue
+from baseplate.lib.message_queue import TimedOutError
+from baseplate.server import make_listener
+from baseplate.server.thrift import make_server
+from baseplate.thrift.message_queue import RemoteMessageQueueService
+from baseplate.thrift.message_queue.ttypes import CreateResponse
+from baseplate.thrift.message_queue.ttypes import GetResponse
+from baseplate.thrift.message_queue.ttypes import PutResponse
+from baseplate.thrift.message_queue.ttypes import ThriftTimedOutError
+
+
+class RemoteMessageQueueHandler:
+    """Create an InMemoryMessageQueue locally and expose get/put methods."""
+
+    def is_healthy(self) -> bool:
+        pass
+
+    def __init__(self) -> None:
+        # Store the queue by name with its max messages
+        self.queues: Dict[str, MessageQueue] = {}
+
+    def create_queue(self, queue_name: str, max_messages: int) -> CreateResponse:
+        queue = InMemoryMessageQueue(queue_name, max_messages)
+        self.queues[queue_name] = queue
+
+        return CreateResponse()
+
+    def get(self, queue_name: str, timeout: Optional[float] = None) -> GetResponse:
+        try:
+            queue = self.queues[queue_name]
+            # Get element from list, waiting if necessary
+            result = queue.get(timeout)
+        # If the queue timed out, raise a timeout as the server response
+        except TimedOutError as e:
+            raise ThriftTimedOutError from e
+
+        return GetResponse(result)
+
+    def put(self, queue_name: str, message: bytes, timeout: Optional[float] = None) -> PutResponse:
+        try:
+            queue = self.queues[queue_name]
```

Comment:
If `queue_name` does not exist, this currently results in an uncaught exception, no?

Reply:
this is true in `get` as well. I waffled about this a little bit - as-is, the queue should always exist, as this handler is only called from a RemoteMessageQueue and RemoteMessageQueue calls create_queue when initialized. I am not sure what it should do in case it doesnt. I could add a new exception type, but that seemed too defensive considering I dont think there is a valid code path that would result in the queue not existing here.